src/PostComment.jsx

```diff
@@ -0,0 +1,36 @@
+import { useState, useContext } from "react";
+import { UserContext } from './contexts/UserContext.jsx'
+import { postComment } from "./utils/newsApi.js";
+
+
+export default function PostComment({ id }) {
+    const { userName } = useContext(UserContext);
+    const [body, setBody] = useState("");
+    const [success, setSuccess] = useState(false)
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+
+        try {
+            console.log(userName);
+            const postedComment = await postComment(id, { username: userName, body });
+            console.log("Comment posted:", postedComment);
+            setSuccess(true)
+
+        } catch (error) {
+            console.error('Error posting comment:', error);
+        }
+    }
```

Comment:
- Remove all console logs before committing work! You shouldn't need to be told this at this point in the week
- For a better user experience the form should clear when a submission has happened. At the moment the text from the submitted comment remains in the input. Please reset this.
- I would consider doing optimistic rendering for the comment posting. Something to explore in your own time. 
- When I throttle my internet to "Slow 3G" and spam the submit button your form allows me to post loads of the same comment. You need to make sure that the form doesn't let the submit button work if the post is still being done. This way a user can't spam the comment form if their internet is slow.
- If I throttle my internet to be "offline" when I submit, I don't get any feedback as a user even though the POST fails. I believe error handling comes up again in task 18 but this directly impacts the use of this form so I'd like you to consider how you can feedback to the user that it's failed.